include/aikido/control/ros/RosJointStateClient.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef AIKIDO_CONTROL_ROS_ROSJOINTSTATECLIENT_HPP_
+#define AIKIDO_CONTROL_ROS_ROSJOINTSTATECLIENT_HPP_
+#include <string>
+#include <boost/circular_buffer.hpp>
+#include <dart/dynamics/dynamics.hpp>
+#include <ros/ros.h>
+#include <ros/callback_queue.h>
+#include <sensor_msgs/JointState.h>
+
+namespace aikido {
+namespace control {
+namespace ros {
+
+/// Client that listens for JointState messages for each skeleton joint and
+/// provides a method for extracting the most recent position of each joint.
+class RosJointStateClient
+{
+public:
+  /// Constructor.
+  /// \param _skeleton Skeleton to read JointState updates for.
+  /// \param _nodeHandle ROS node.
+  /// \param _topicName Name of topic to subscribe to for JointState updates.
+  /// \param _capacity Number of JointStateRecords that are saved per joint.
+  RosJointStateClient(
+    dart::dynamics::SkeletonPtr _skeleton,
+    ::ros::NodeHandle _nodeHandle,
+    const std::string& _topicName,
+    size_t capacity);
+
+  /// Update mBuffer with any JointState messages that have been received.
+  void spin();
+
+  /// Returns the most recent position of each joint in _metaSkeleton.
+  /// \param _metaSkeleton Skeleton to read DOFs from.
+  /// \return vector of positions for each DOF
+  Eigen::VectorXd getLatestPosition(
+    const dart::dynamics::MetaSkeleton& _metaSkeleton) const;
+
+  // TODO: implement
+  // getPositionAtTime(const MetaSkeleton&, const ros::Time&, bool)
+  // that interpolates position at the specified time, optionally blocking for
+  // new data.
+private:
+  struct JointStateRecord
+  {
+    inline bool isValid() const { return mStamp.isValid(); }
```

Comment:
From my understanding, this is not true. Do you have a citation for that? :neckbeard: 

The `inline` keyword is a _suggestion_ that the compiler should inline the function that it can choose to honor or ignore, depending upon what it's own optimizer thinks. It's true that GCC and Clang both aggressively inline functions that are available in the current translation unit (e.g. those that are defined in headers), but the `inline` keyword may still have an effect.